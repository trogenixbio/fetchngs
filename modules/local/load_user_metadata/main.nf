
process LOAD_USER_METADATA {
    tag "$metadata"

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-693e24f156d01a5f55647120be99929b01b30949:609c862c3470382215fc1b2d9d8a4e9637b2e25f-0' :
        'biocontainers/mulled-v2-693e24f156d01a5f55647120be99929b01b30949:609c862c3470382215fc1b2d9d8a4e9637b2e25f-0' }"

    input:
    path input_metadata
    val is_excel

    output:
    path "metadata.json", emit: metadata

    script:
    def is_excel = is_excel ? "--is_excel": ""
    """
    load_user_metadata.py $is_excel $input_metadata metadata.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS

    """
}
