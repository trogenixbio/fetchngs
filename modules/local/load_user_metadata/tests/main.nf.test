nextflow_process {

    name "Test Process LOAD_USER_METADATA"
    script "../main.nf"
    process "LOAD_USER_METADATA"

    test("Load in xlsx") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "metadata_template_ERC000011_TRX.xlsx", checkIfExists: true)
                input[1] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Load in xlsx meta") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "metadata_template_ERC000011_TRX_Meta.xlsx", checkIfExists: true)
                input[1] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Load in TSV set") {

        when {
            process {
                """
                ch_tsv = Channel.fromPath( ["$baseDir" + "/" + params.modules_testdata_base_path_local + "study.tsv",
                                            "$baseDir" + "/" + params.modules_testdata_base_path_local + "experiment.tsv",
                                            "$baseDir" + "/" + params.modules_testdata_base_path_local + "sample.tsv",
                                            "$baseDir" + "/" + params.modules_testdata_base_path_local + "run.tsv"] )
                input[0] = ch_tsv.collect()
                input[1] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Load in TSV set no metadata") {

        when {
            process {
                """
                ch_tsv = Channel.fromPath( ["$baseDir" + "/" + params.modules_testdata_base_path_local + "load_user_metadata/study.tsv",
                                            "$baseDir" + "/" + params.modules_testdata_base_path_local + "load_user_metadata/experiment.tsv",
                                            "$baseDir" + "/" + params.modules_testdata_base_path_local + "load_user_metadata/sample.tsv",
                                            "$baseDir" + "/" + params.modules_testdata_base_path_local + "load_user_metadata/run.tsv"] )
                input[0] = ch_tsv.collect()
                input[1] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
