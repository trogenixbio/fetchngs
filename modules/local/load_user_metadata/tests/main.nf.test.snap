{
    "Load in xlsx": {
        "content": [
            {
                "0": [
                    "metadata.json:md5,66957dbc73243ee1a5506597a26f8eb5"
                ],
                "metadata": [
                    "metadata.json:md5,66957dbc73243ee1a5506597a26f8eb5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-22T14:11:06.917073694"
    },
    "Load in xlsx meta": {
        "content": [
            {
                "0": [
                    "metadata.json:md5,d8bfe7111fc8fb5dcfa4ac3fe68e6a53"
                ],
                "metadata": [
                    "metadata.json:md5,d8bfe7111fc8fb5dcfa4ac3fe68e6a53"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-22T14:12:36.337115882"
    },
    "Load in TSV set": {
        "content": [
            {
                "0": [
                    "metadata.json:md5,0744878eaf61572aa7d4355a69cd77ea"
                ],
                "metadata": [
                    "metadata.json:md5,0744878eaf61572aa7d4355a69cd77ea"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-22T14:11:16.820558696"
    }
}